subprojects/build-init/src/main/java/org/gradle/api/tasks/wrapper/Wrapper.java

```diff
@@ -169,7 +170,7 @@ private void writeProperties(File propertiesFileDestination) {
      */
     @OutputFile
     public File getScriptFile() {
-        return getProject().file(scriptFile);
+        return getServices().get(FileOperations.class).file(scriptFile);
```

Comment:
I preferred not to in those cases because `FileOperations` is internal and injecting would mean changing the constructor signature or adding a protected function, exposing the internal type.

Reply:
That means we need a public service that could replace it, right?

Reply:
This use case is one of the problematic ones yes.
https://github.com/gradle/gradle/issues/11774

Reply:
Ok.